Starcounter.Uniform.Tests/Queryables/QueryableFilterTests.cs

```diff
@@ -54,5 +55,129 @@ public void ApplyPartialFilterShouldReturnProperRowDatas()
             returnedData.Should().HaveCount(2);
             returnedData.Should().NotContain(x => x.Name == "Tom");
         }
+
+        [Test]
+        public void ApplyingMultipleFiltersShouldReturnProperRowsDatas()
+        {
+            var data = new List<RowDataModel>
+            {
+                new RowDataModel {Name = "Ann", Number = 0},
+                new RowDataModel {Name = "Amanda", Number = 1},
+                new RowDataModel {Name = "Tom", Number = 2},
+                new RowDataModel {Name = "Clark", Number = 3},
+            }.AsQueryable();
+            var aFilter = new Filter { PropertyName = "Name", Value = "A" };
+            var annFilter = new Filter { PropertyName = "Name", Value = "Ann" };
+            var filterOrderConfiguration = new FilterOrderConfiguration();
+            filterOrderConfiguration.Filters.Add(aFilter);
+            filterOrderConfiguration.Filters.Add(annFilter);
+
+            var returnedData = _sut.Apply(data, filterOrderConfiguration);
+
+            returnedData.Should().ContainSingle().Which.Name.Should().Be("Ann");
+        }
+
+        [Test]
+        public void ApplyingOrderingShouldReturnProperRowsDatasOrdered()
+        {
+            var data = new List<RowDataModel>
+            {
+                new RowDataModel {Name = "Tom", Number = 2},
+                new RowDataModel {Name = "Ann", Number = 0},
+                new RowDataModel {Name = "Clark", Number = 2},
+                new RowDataModel {Name = "Amanda", Number = 1}
+            }.AsQueryable();
+
+            var filterOrderConfiguration = new FilterOrderConfiguration()
+            {
+                Ordering = {new Order {PropertyName = nameof(RowDataModel.Name), Direction = OrderDirection.Ascending}}
+            };
+
+            var returnedData = _sut.Apply(data, filterOrderConfiguration);
+
+            returnedData.Select(model => model.Name).Should().BeInAscendingOrder();
+        }
+
+        [Test]
+        public void ApplyingIntOrderingShouldReturnProperRowsDatasOrdered()
+        {
+            var data = new List<RowDataModel>
+            {
+                new RowDataModel {Name = "Tom", Number = 2},
+                new RowDataModel {Name = "Ann", Number = 0},
+                new RowDataModel {Name = "Clark", Number = 3},
+                new RowDataModel {Name = "Amanda", Number = 1}
+            }.AsQueryable();
+            var filterOrderConfiguration = new FilterOrderConfiguration()
+            {
+                Ordering =
+                {
+                    new Order {PropertyName = nameof(RowDataModel.Number), Direction = OrderDirection.Ascending}
+                }
+            };
+
+            var returnedData = _sut.Apply(data, filterOrderConfiguration);
+
+            returnedData.Select(model => model.Number).Should().BeInAscendingOrder();
+        }
+
+        [Test]
+        public void ApplyingMultipleOrderingShouldReturnProperRowsDatasOrdered()
```

Comment:
does this really check the sorting? `Amanda` comes before `Ann` in the input set. It looks like it just ordered by number

Reply:
If we call ordering for first Names than numbers I assume that it should be ordered first by names than by numbers. But now when I think of it, this case should be visible with 2 same entries but different numbers. I'll edit data to add this case.